shared/nav.desktop.js

```diff
@@ -163,6 +164,10 @@ class Nav extends Component<void, Props, State> {
   }
 
   shouldComponentUpdate (nextProps, nextState) {
+    if (this.props.folderBadge !== nextProps.folderBadge) {
```

Comment:
react will soon have a Component.pure we could derive from. in cases like this where we have a highly specific shouldComponentUpdate we just have to watch out for this. a side effect of this kind of optimization